sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/jwt/AadJwtDecoderProviderConfiguration.java

```diff
@@ -37,45 +36,37 @@ private AadJwtDecoderProviderConfiguration() {
-        return getConfiguration(oidcIssuerLocation, oidc(URI.create(oidcIssuerLocation)));
-    }
-
-    private static Map<String, Object> getConfiguration(String issuer, URI... uris) {
-        String errorMessage = "Unable to resolve the Configuration with the provided Issuer of " + issuer;
-
-        for (URI uri : uris) {
-            try {
-                RequestEntity<Void> request = RequestEntity.get(uri).build();
-                ResponseEntity<Map<String, Object>> response = REST.exchange(request,
-                    TYPE_REFERENCE);
-                Map<String, Object> configuration = response.getBody();
-                if (configuration == null) {
-                    throw new IllegalArgumentException("The configuration must not be null");
-                }
-                if (configuration.get("jwks_uri") == null) {
-                    throw new IllegalArgumentException("The public JWK set URI must not be null");
-                }
+    static Map<String, Object> getConfigurationForOidcIssuerLocation(RestOperations restOperations, String oidcIssuerLocation) {
+        URI issuer = URI.create(oidcIssuerLocation);
+        String errorMessage = "Unable to resolve the Configuration with the provided Issuer of " + oidcIssuerLocation;
+        URI uri = UriComponentsBuilder.fromUriString(oidcIssuerLocation)
+                                      .replacePath(issuer.getPath() + OIDC_METADATA_PATH)
+                                      .build(Collections.emptyMap());
+        try {
+            RequestEntity<Void> request = RequestEntity.get(uri).build();
+            ResponseEntity<Map<String, Object>> response = restOperations.exchange(request,
+                TYPE_REFERENCE);
+            Map<String, Object> configuration = response.getBody();
+            if (configuration == null) {
+                throw new IllegalArgumentException("The configuration must not be null");
+            }
+            if (configuration.get("jwks_uri") == null) {
+                throw new IllegalArgumentException("The public JWK set URI must not be null");
+            }
 
-                return configuration;
-            } catch (IllegalArgumentException e) {
-                throw e;
-            } catch (RuntimeException e) {
-                if (!(e instanceof HttpClientErrorException
-                    && ((HttpClientErrorException) e).getStatusCode().is4xxClientError())) {
-                    throw new IllegalArgumentException(errorMessage, e);
-                }
-                // else try another endpoint
+            return configuration;
+        } catch (IllegalArgumentException e) {
+            throw e;
+        } catch (RuntimeException e) {
+            if (!(e instanceof HttpClientErrorException
+                && ((HttpClientErrorException) e).getStatusCode().is4xxClientError())) {
+                throw new IllegalArgumentException(errorMessage, e);
             }
+            // else try another endpoint
```

Comment:
I think `AadJwtDecoderProviderConfiguration` is copied from [JwtDecoderProviderConfigurationUtils](https://github.com/spring-projects/spring-security/blob/5.7.2/oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/JwtDecoderProviderConfigurationUtils.java). 

`else try another endpoint` is confusing because current `AadJwtDecoderProviderConfiguration` doesn't have `other endpoint`.